src/main/drivers/timer_hal.c

```diff
@@ -256,7 +256,7 @@ void configTimeBase(TIM_TypeDef *tim, uint16_t period, uint8_t mhz)
     timerHandle[timerIndex].Handle.Instance = tim;
 
     timerHandle[timerIndex].Handle.Init.Period = (period - 1) & 0xffff; // AKA TIMx_ARR
-    timerHandle[timerIndex].Handle.Init.Prescaler = (SystemCoreClock / timerClockDivisor(tim) / ((uint32_t)mhz * 1000000)) - 1;
+    timerHandle[timerIndex].Handle.Init.Prescaler = (timerClock(tim) / ((uint32_t)mhz * 1000000)) - 1;
```

Comment:
Explicit cast to `uint32_t` is not necessary as `100000` literal type is `int` and `mhz` will be promoted according to integral promotion rules.
As you're doing a PR on this one, we may as well drop the cast :-)